contracts/PSM/BasePSM.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { Ownable2Step } from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import { Fixed256x18 } from "@tempusfinance/tempus-utils/contracts/math/Fixed256x18.sol";
+import { ERC20PermitSignature } from "@tempusfinance/tempus-utils/contracts/utils/PermitHelper.sol";
+import { IRToken } from "../Interfaces/IRToken.sol";
+import { IPositionManager } from "../Interfaces/IPositionManager.sol";
+import { IPSM } from "./IPSM.sol";
+import { IPSMFeeCalculator } from "./IPSMFeeCalculator.sol";
+
+/// @dev Base implementation of Peg Stability Module.
+/// Handles basic reserve token transfers as well as minting R.
+abstract contract BasePSM is IPSM, ERC20, Ownable2Step {
+    using SafeERC20 for IERC20;
+    using Fixed256x18 for uint256;
+
+    IERC20 public immutable override reserveToken;
+    IRToken public immutable override rToken;
+    IPositionManager public immutable override positionManager;
+    IPSMFeeCalculator public override feeCalculator;
+
+    constructor(
+        IERC20 reserveToken_,
+        IRToken rToken_,
+        string memory name_,
+        string memory symbol_,
+        IPSMFeeCalculator feeCalculator_
+    )
+        ERC20(name_, symbol_)
+    {
+        reserveToken = reserveToken_;
```

Comment:
Maybe check that `reserveToken_` is not a zero address?

Reply:
Added